src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -286,6 +312,10 @@ public Command OnErrorLine(Action<string> handler)
             return this;
         }
 
+        public CommandResolutionStrategy CommandResolution { get; }
+
+        public string CommandName => _process.StartInfo.FileName;
```

Comment:
I also wondered what to put there. `_process.StartInfo.FileName` represents the command string that actually gets resolved, which might differ from the user specified command. I considered former would be more helpful to put in the precondition message than the latter.

Would you rather have the user specific string there?